MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoReaderUtils.java

```diff
@@ -0,0 +1,143 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.android.libraries.motionphotoreader.Constants.NUM_OF_STRIPS;
+import static com.google.android.libraries.motionphotoreader.Constants.TIMEOUT_US;
+
+/**
+ * A collection of helper methods used by MotionPhotoReader to access buffers and read track data.
+ */
+@RequiresApi(api = 28)
+class MotionPhotoReaderUtils {
+    private static final String TAG = "MotionPhotoReaderUtils";
+
+    /**
+     * Retrieve the index of the next available input buffer.
+     * @param availableInputBuffers A blocking queue containing the indices of free input buffers.
+     * @return the index of the next available input buffer, or null if the poll call results in a
+     * timeout.
+     */
+    public static Integer getInputBuffer(BlockingQueue<Integer> availableInputBuffers) {
+        Integer bufferIndex = -1;
+        try {
+            bufferIndex = availableInputBuffers.poll(TIMEOUT_US, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            Log.e(TAG, "No input buffer available", e);
```

Comment:
In this code path, we return `-1` which is undocumented by Javadoc. Should we also return the special value `null` in that case?

Reply:
Right, I think it should be null (made a change to initialize bufferIndex as null).